src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityEventSummary.kt

```diff
@@ -62,6 +174,128 @@ object HoppityEventSummary {
+
+        // If we're only showing the live display during the last {X} hours of the hunt,
+        // check if we're in that time frame
+        if (updateCfConfig.showForLastXHours > 0) {
+            val eventEndTm = HoppityAPI.millisToEventEnd().asTimeMark()
+            if (eventEndTm.timeUntil() >= updateCfConfig.showForLastXHours.hours) return
+        }
+
+        // If it's been less than {config} minutes since the last message, don't send another
+        lastSentCfUpdateMessage.let {
+            val configFrequency = updateCfConfig.reminderInterval
+            if (it != null && it.passedSince() < configFrequency.minutes) return
+        }
+        val stats = getYearStats().first ?: return
+        val markDiff = stats.lastLbUpdateMarkMillis.asTimeMark().passedSince()
+
+        // If it's been more than {config} since the last update, send a message
+        if (markDiff >= updateCfConfig.reminderInterval.minutes) {
+            lastSentCfUpdateMessage = SimpleTimeMark.now()
+            ChatUtils.chat(
+                "§6§lReminder! §r§eSwitch to a new server and run §6/cf §e to " +
+                    "update your leaderboard position in Hoppity Event stats."
+            )
+        }
+    }
+
+    private fun buildDisplayRenderables(stats: HoppityEventStats?, statYear: Int): List<Renderable> = buildList {
+        // Add title renderable with centered alignment
+        add(
+            Renderable.string(
+                "§dHoppity's Hunt #${getHoppityEventNumber(statYear)} Stats",
+                horizontalAlign = RenderUtils.HorizontalAlignment.CENTER
+            )
+        )
+
+        // Conditionally add year switcher renderable for inventory or chest screens
+        if (isInInventory()) {
+            buildYearSwitcherRenderables(statYear)?.let { yearSwitcher ->
+                add(
+                    Renderable.horizontalContainer(
+                        yearSwitcher,
+                        spacing = 5,
+                        horizontalAlign = RenderUtils.HorizontalAlignment.CENTER
+                    )
+                )
+            }
+        }
+
+        // Add card renderable based on stats availability
+        val cardRenderable = if (stats == null) {
+            Renderable.verticalContainer(
+                mutableListOf(Renderable.string("§cNo stats found for Hunt #${getHoppityEventNumber(statYear)}."))
+            )
+        } else {
+            Renderable.verticalContainer(
+                getStatsStrings(stats, statYear).map { Renderable.string(it.string) }.toMutableList()
+            )
+        }
+        add(cardRenderable)
+    }
+
+    private fun buildYearSwitcherRenderables(currentStatYear: Int): List<Renderable>? {
+        val profileStorage = ProfileStorageData.profileSpecific ?: return null
+        val statsStorage = profileStorage.hoppityEventStats
+        val statsYearList = statsStorage.keys.takeIf { it.isNotEmpty() } ?: mutableListOf()
```

Comment:
why exactly do you need the isNotEmpty check? all you are doing here is "if its not empty, use it, if its empty create a new empty list"?